custom/enikshay/two_b_release_1/management/commands/enikshay_2b_case_properties.py

```diff
@@ -0,0 +1,496 @@
+            episodes = [case for case in episodes
+                        if not case.closed and case.type == CASE_TYPE_EPISODE
+                        and any([index.referenced_id == occurrence.case_id for index in case.indices])
+                        and case.dynamic_case_properties().get('episode_type') == "confirmed_tb"]
+            episode = episodes[0] if episodes else None
+        else:
+            episode = None
+
+        props = {
+            'enrolled_in_private': 'false',
+            'case_version': '20',
+            'area': person.get_case_property('phi_area'),
+            'language_code': 'hin',
+            'referred_outside_enikshay_date': person.get_case_property('date_referred_out'),
+            'referred_outside_enikshay_by_id': person.get_case_property('date_by_id'),
+        }
+        if episode:
+            props.update({
+                'current_episode_type': episode.get_case_property('episode_type'),
+                'alcohol_history': episode.get_case_property('alcohol_history'),
+                'alcohol_deaddiction': episode.get_case_property('alcohol_deaddiction'),
+                'tobacco_user': episode.get_case_property('tobacco_user'),
+                'occupation': episode.get_case_property('occupation'),
+                'nikshay_id': episode.get_case_property('nikshay_id'),
+                'phone_number_other': episode.get_case_property('phone_number_other'),
+            })
+
+        phone_number = person.get_case_property('phone_number')
+        if phone_number:
+            # TODO check if it already begins with 91 or has a certain length?
+            props['contact_phone_number'] = '91' + phone_number
+
+        location = self.locations.get(person.owner_id)
+        if location:
+            if location.location_type.code == 'phi':
+                props['phi_name'] = location.name
+            ancestors_by_type = self.get_ancestors_by_type(location)
+            if 'tu' in ancestors_by_type:
+                props['tu_name'] = ancestors_by_type['tu'].name
+                props['tu_id'] = ancestors_by_type['tu'].location_id
+            if 'dto' in ancestors_by_type:
+                props['dto_name'] = ancestors_by_type['dto'].name
+                props['dto_id'] = ancestors_by_type['dto'].location_id
+
+        return CaseStructure(
+            case_id=person.case_id,
+            walk_related=False,
+            attrs={
+                "create": False,
+                "owner_id": person.owner_id or '-',
+                "update": props,
+            },
+        )
+
+    def migrate_occurrence(self, occurrence, episodes):
+        self.total_occurrences += 1
+        episodes = [(case.opened_on, case) for case in episodes
+                    if case.type == CASE_TYPE_EPISODE
+                    and any([index.referenced_id == occurrence.case_id for index in case.indices])
+                    and case.dynamic_case_properties().get('episode_type') == "confirmed_tb"]
```

Comment:
Same comment as above - is this also the case when retrieving the most recently opened episode for the occurrence?